vault/token_store.go

```diff
@@ -362,13 +363,6 @@ func (ts *TokenStore) paths() []*framework.Path {
 				OperationVerb:   "look-up",
 			},
 
-			Fields: map[string]*framework.FieldSchema{
-				"token": {
-					Type:        framework.TypeString,
-					Description: "Token to look up (unused, does not need to be set)",
-				},
-			},
-
```

Comment:
Note for reviewer: This field is being removed because it is unused from the user perspective, and even documented as such, and if I did not remove it, I'd need to add `Query: true` to it, to satisfy the heuristic I was using to find all these fields. (I used Go's reflect package to find all of the places where the same callback function was shared between read-type and update-type operations, not already marked Query: true.)

The reason this field is actually here is to make the request schema for `auth/token/lookup-self` match the schema for `auth/token/lookup`, so that `handleLookupSelf` can be implemented as an awkward hack that stuffs the request token into the internal request data, and forwards directly to the code that handles `auth/token/lookup`. There's another related awkward hack used to implement `auth/token/lookup-accessor` too. All the rest of the change in this file is to do with fixing those hacks to use nicer code.

As I look over this PR the following day, I feel like I might have strayed over the line into mixing too much opportunistic code cleanup into this change. I would be happy to move this change and all following it in this file, into a separate cleanup PR.

Reply:
This make sense, but probably best to move the code refactor & cleanup it into its own PR as you mentioned :)